File: ProjectEuler/1-10/projectEuler6.py
"""
Description: Find the absolute difference between the sum of the squares of the
first natural numbers and the square of the sum. I used the 
mathematical sigma notation for sum of squares of n natural numbers:
n(n*1)(2n+1)/6. Completed using Python 3 
"""

#!/bin/python3

import sys

def sumSquareDifference(n):
    squareOfSumOfNaturalNumbers = 0 
    sumOfSquares = ((n) * (n + 1) * (2*n + 1)) / 6 
    for i in range(1, n + 1):
        squareOfSumOfNaturalNumbers += i 
    squareOfSumOfNaturalNumbers = squareOfSumOfNaturalNumbers**2 
    return abs((int) (squareOfSumOfNaturalNumbers - sumOfSquares))

t = int(input().strip())
for a0 in range(t):
    n = int(input().strip())
    print(sumSquareDifference(n))

File: ProjectEuler/1-10/project-euler-6.java
/*
Description: Find the absolute difference between the sum of the squares of the
first natural numbers and the square of the sum. Completed using Java 8 
*/

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static long sumSquareDifference(long n) 
    {
        long squareOfSumOfNaturalNumbers = 0, sumOfSquares = 0; 
        for(long i = 1; i < n + 1; i++)
        {
            squareOfSumOfNaturalNumbers += i; 
            sumOfSquares += Math.pow(i, 2); 
        }
        squareOfSumOfNaturalNumbers = 
           (long) Math.pow(squareOfSumOfNaturalNumbers, 2); 
        return Math.abs(squareOfSumOfNaturalNumbers - sumOfSquares); 
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            System.out.println((long) sumSquareDifference(n)); 
        }
    }
}